src/lando/api/legacy/workers/landing_worker.py

```diff
@@ -271,13 +272,27 @@ def run_job(self, job: LandingJob) -> bool:
 
             # Run through the patches one by one and try to apply them.
             for revision in job.revisions.all():
-                patch_buf = StringIO(revision.patch_string)
 
                 try:
-                    repo.hg.apply_patch(patch_buf)
+                    # TODO: Rather than parsing the patch details from the full HG patch
+                    # stored in the job, we should read the revision's metadata (and
+                    # move to only store the diff in the patch_string, rather than an
+                    # export).
+                    patch_helper = HgPatchHelper(StringIO(revision.patch_string))
+                    if not patch_helper.diff_start_line:
+                        raise NoDiffStartLine()
```

Comment:
This can be cleaned up by moving everything in lines 292-304 into this if statement, and moving the if statement outside the try block.